### DISCUSSION TITLE
module path for HAFS does not exist

### DISCUSSION THREAD

**Discussion by DanWu85** [hafs-community/HAFS#225] [Q&A] [answered]
Hi,

This is Dan from HRD. I'm running HAFS on orion. However, I had trouble submitting jobs because it could not find the module needed.  Folks in EMC and OU who are working on HAFS project have the same issue.

prepend_path("MODULEPATH", "/work/noaa/epic-ps/hpc-stack/libs/intel/2022.1.2/modulefiles/stack")

I've learned that the project 'epic-ps' was closed by the Pi.  The folder no longer exists on the scratch drive for /work or /work2. 

Is it possible to reinstall the libraries or packages used by hafs on some shared place? Many thanks!

Best,
Dan

**Comment by natalie-perlin** [comment, marked as answer]
Hi Dan,
   Please note that the space on Orion  **/work/noaa/epic-ps/** is no longer available, and the need to migrate to a new location for EPIC was requested earlier this year. All the data and stacks maintained by EPIC have been rebuild or moved to a new EPIC-dedicated space under **/work/noaa/epic/role-epic/**. The UFS WM and UFS-SRW App communities have transitioned to use this new project space, and I'm sorry to hear the communication with the HAFS team has missed this issue. 
   
The following software stack built with the same intel/2022.1.2 compiler on Orion could be tested now to load modules for HAFS (changes to the [hafs.orion.lua](https://github.com/hafs-community/HAFS/blob/develop/modulefiles/hafs.orion.lua) modulefile):
    
```
prepend_path("MODULEPATH", "/work/noaa/epic/role-epic/contrib/orion/hpc-stack/intel-2022.1.2/modulefiles/stack")
```

Two more changes are needed to use different software versions for fms/2022.04 and bufr/11.7.0:

```
fms_ver=os.getenv("fms_ver") or "2022.04"
load(pathJoin("fms",fms_ver))
```

and 

```
bufr_ver=os.getenv("bufr_ver") or "11.7.0"
load(pathJoin("bufr", bufr_ver))
```

Please let us know whether these changes resolved the issues and if all works as expected with the new stack!
    
Respectfully,
    Natalie.

**Reply by XuLu-NOAA** [reply]
Hi, Natalie,

This is Xu from OU. I've tested the stack from role-epic, it went fine for the HAFS forecast jobs. But I've got errors when running GSI and unipost jobs because of the missing paths for crtm2.4.0.  They are currently still linked from epic-ps dir.
Would you mind also fixing this lib/module?

Thanks,
Xu

**Comment by natalie-perlin** [comment]
Dear Xu, 
  
GSI and unipost jobs need to have the path adapted to a new location as well. Are GSI and unipost are external submodules of HAFS, do they need to have the path for CRTM adjusted in their separate repositories?

**Reply by XuLu-NOAA** [reply]
Thanks, the problem is resolved!

**Comment by natalie-perlin** [comment]
It sounds like issues that Xu reported have been now resolved.
